ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/MainnetGenesisFileModule.java

```diff
@@ -31,14 +41,96 @@ class MainnetGenesisFileModule extends GenesisFileModule {
   }
 
   @Override
-  ProtocolSchedule provideProtocolSchedule(
+  BlockHeaderFunctions blockHashFunction() {
+    return new MainnetBlockHeaderFunctions();
+  }
+
+  @Override
+  HeaderBasedProtocolSchedule provideProtocolSchedule(
       final GenesisConfigOptions configOptions,
+      @Named("Fork") final Optional<String> fork,
       @Named("RevertReasonEnabled") final boolean revertReasonEnabled) {
+    if (fork.isPresent()) {
+      var schedules = createSchedules();
+      var schedule = schedules.get(fork.map(String::toLowerCase).get());
+      if (schedule != null) {
+        return schedule.get();
+      }
+    }
     return MainnetProtocolSchedule.fromConfig(configOptions, EvmConfiguration.DEFAULT);
   }
 
-  @Override
-  BlockHeaderFunctions blockHashFunction() {
-    return new MainnetBlockHeaderFunctions();
+  public static Map<String, Supplier<HeaderBasedProtocolSchedule>> createSchedules() {
```

Comment:
This looks like a new place to remember to update when we create new forks.  I wish we had a way of discovering these places without resorting to some kind of rigid static analysis/annotation

Reply:
The main issue here is that it crosses module barriers.  EVM has configurations that need to be set up, and so does the mainnet client.  Throw in if we do things like separate out the block and tx data validation that those also have different rules based on different forks.

But if we had all EVM changes come from one source that would be good.  Will think about it once EOF settles down.